### ISSUE TITLE
question: when adding a cookie re: the notify-send, what needs to occur to display?

### ISSUE THREAD

**Issue by borisivan** [zalando/zalenium#268]
Hi,

I've added support in my tests to show the 'test failed' based on adding the cookie, and it works... but shows up quite a bit later than when I add it. I'm not sure if it's simply time based, or based on the next RemoteWebDriver event that occurs after I've added the cookie.

For example, I really want to identify precisely when a test fails (as close as possible). When I've concluded it fails, I add the cookie, and I let the browser stay open for 5 more seconds before closing it. But the notify only shows when I _close_ the browser, which was 5 seconds after I added the cookie.

If it's based on some additional action needing to occur, can you recommend a webDriver action that we can do that might flush this? I don't want to do a 'screen refresh' or anything like that, as I want the browser to remain in the exact state it was in, when I conclude the test fails.

Thanks!

**Comment by borisivan** [comment]
May have figured it out -- am I correct in saying that the test failed message is treated special, and only displayed once the browser is closed?

For 'zaleniumMessage', it seems to display instantly. But for zaleniumTestPassed, it doesn't seem to display when I add that cookie, but it does when I close the browser. Is that correct?

**Comment by diemol** [comment]
Hi @borisivan,

Exactly, the message that says the final test status is part of the final activities, which are triggered when the "delete session" command is received. That is why the notification is always seen after the browser is closed.

The behaviour of  "zaleniumMessage" is different, is shows up right away. So it is working as it is supposed to be.

If you want to identify the exact moment, I would suggest to use the "zaleniumMessage" to mark the steps of your test, and then the last one shown should be the step where the step failed.

**Comment by borisivan** [comment]
Yep, I transitioned to using the 'message' for right at the instant I conclude the test fails, so it's shown in the video at the exact time. All set, thx.

**Comment by diemol** [comment]
Thanks for the feedback @borisivan!
